### ISSUE TITLE
`Gen<_>.fromElements(of:)` does not accept sets

### ISSUE THREAD

**Issue by dented42** [typelift/SwiftCheck#238]
Version
========

912f0f6f53067a1232c07dc5ecffe05cdfbc1e5d (swift-develop branch)

Environment
===========

MacOS 10.12.6 with Xcode 9 beta 6

Description
============

`Set` cannot be used as an argument to `Gen<_>.fromElements(of:)`.

Steps To Reproduce
===================

```
let s: Set<Int> = Set([2,3,5,7])
let sGen: Gen<Int> = Gen.fromElements(of: s)
```

Expected Result
================

`sGen` should be a generator for single digit prime numbers.

Actual Result
==============

Xcode firstly complains 'incorrect argument label in call (have 'of:', expected 'in:')`.

Additional information
======================

A work around is to make an array out of the set and pass that to `.fromElements(of:)`. This however is rather ugly.

**Comment by CodaFi** [comment]
This is an awful diagnostic that's emitted because `Set.Index` is its own custom type now and needs a `RandomType` instance.

**Comment by CodaFi** [comment]
In the meantime, this should do it.

```swift
extension Gen {
	/// Constructs a Generator that selects a random value from the given
	/// collection and produces only that value.
	///
	/// The input collection is required to be non-empty.
	public static func fromElements<T>(of xs : Set<T>) -> Gen<T> {
		precondition(!xs.isEmpty)
		return Gen.fromElements(in: 0...xs.distance(from: xs.startIndex, to: xs.endIndex)-1).map { i in
			return xs[xs.index(xs.startIndex, offsetBy: i)]
		}
	}
}
```

**Comment by CodaFi** [comment]
This overload was added in #237 and shipped with 0.8.1.  Closing.
